src/reducers/sources.js

```diff
@@ -218,6 +221,27 @@ function updateTabList(state: OuterState, url: ?string, tabIndex?: number) {
   return tabs;
 }
 
+function updateBlackBoxList(url, isBlackBoxed) {
+  let tabs = getBlackBoxList();
+  const i = tabs.indexOf(url);
+  if (i >= 0) {
+    if (!isBlackBoxed) {
+      tabs.splice(i, 1);
+    }
+  } else if (isBlackBoxed) {
+    tabs.push(url);
+  }
+  prefs.tabsBlackBoxed = tabs;
+}
+
+export function getBlackBoxList() {
+  let tabs = prefs.tabsBlackBoxed;
+  if (!tabs) {
+    tabs = [];
+  }
+  return tabs;
```

Comment:
This function could just be `return prefs.tabsBlackBoxed || [];`

Reply:
Updated, thanks